nodes/persistence-node/src/services/IpfsGatewayApi.ts

```diff
@@ -88,7 +89,14 @@ export class IpfsGatewayApi {
     app.get('/api/v0/resolve', handleError(async (req, res) => {
       const hash = req.query.arg as string;
 
-      const resolvedPath = await ipfs.resolve(`/ipfs/${hash}`);
+      const resolvedPath = await ipfs.resolve(`/ipfs/${hash}`)
+        .catch(err => {
```

Comment:
Should we create new wrapper service around ipfs to map these errors to our custom errors, to keep controllers clean?

Reply:
You could create something in `ipfs-operations` directory. For error handling, instead of throwing errors, you could follow the pattern of returning it in `[error, result]` format like here:

https://github.com/polywrap/nodes/pull/66/files#diff-8e61310f070187bc789209ef7decbf9c223e001be7937f71ccfdeae02968c7e7R78

Reply:
If `[error, result]` is a convention we follow, sure, I'll change it to that.

Reply:
@nerfZael What are your thoughts on `[error, result]` format.
I personally find it more cumbersome than throwing an error.

1. We always have to return tuple where one element is null/undefined and other is present.
2. We are always forced to handle error at call site, polluting happy case path, where we usually just want to abort in the end anyway.
3. If error type is string, we don't have robust control on what to do on call site. To have control, we should either define type literal strings or create some error type, and all of this is already present out of the box when throwing errros.
4. We already have error handler middleware where we can handle all custom errors equally.
5. If we want to handle error immediately, we can still write try/catch at call site

Reply:
I think the best way would be to throw errors (If you don't need to worry about them) or if inside of standard functions, while having a helper function to wrap others and help handle the errors (because try catch is awful)
Something like this function which has been used on another project:
```
export const handlePromiseError = async <TError, TData>(
  promise: Promise<TData>
): Promise<[TError | undefined, TData | undefined]> => {
  let error: TError | undefined = undefined;
  let data: TData | undefined = undefined;

  const result = await promise.catch((err: TError) => {
    error = err;
  });

  data = result as TData | undefined;

  return [error, data];
};

```

Reply:
I threw error.

This util function is nice in case when we want to handle error directly.